content/en/blog/_posts/2023-04-11-topology-spread-features.md

```diff
@@ -0,0 +1,140 @@
+because we didn't take care of node taints, which will lead to the pod pending.
+
+To avoid this and make a more fine-gained decision in scheduling, we introduced two new fields in
+`TopologySpreadConstraint` to define node inclusion policies including nodeAffinity and nodeTaint.
+
+It mostly looks like:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example-pod
+spec:
+  # Configure a topology spread constraint
+  topologySpreadConstraints:
+    - maxSkew: <integer>
+      # ...
+      nodeAffinityPolicy: [Honor|Ignore]
+      nodeTaintsPolicy: [Honor|Ignore]
+  # other Pod fields go here
+```
+
+**nodeAffinityPolicy** indicates how we'll treat Pod's nodeAffinity/nodeSelector in pod topology spreading.
+If `Honor`, we'll filter out nodes not matching nodeAffinity/nodeSelector in calculation.
+If `Ignore`, these nodes will be included instead.
+
+For backwards-compatibility, nodeAffinityPolicy is default to `Honor`.
+
+**nodeTaintsPolicy** indicates how we'll treat node taints in pod topology spreading.
+If `Honor`, only tainted nodes for which the incoming pod has a toleration, will be included in calculation.
+If `Ignore`, we'll not consider the node taints at all in calculation, so a node with pod untolerated taint
+will also be included.
+
+For backwards-compatibility, nodeTaintsPolicy is default to the `Ignore`.
+
+The feature was introduced in v1.25 as alpha level. By default, it was disabled, so if you want to use this feature in v1.25,
+you have to enable the feature gate `NodeInclusionPolicyInPodTopologySpread` actively. In the following v1.26, we graduated
+this feature to beta and it was enabled by default since.
+
+## KEP-3243: Respect PodTopologySpread after rolling upgrades
+
+Pod Topology Spread uses the fields `topologyKey` or `labelSelector` to identify the group of pods over which
+spreading will be calculated. But it applies to all pods in a Deployment irrespective of their owning
+ReplicaSet. As a result, when a new revision is rolled out, spreading will apply across pods from both the
+old and new ReplicaSets, and so by the time the new ReplicaSet is completely rolled out and the old one is
+rolled back, the actual spreading we are left with may not match expectations because the deleted pods from
+the older ReplicaSet will cause skewed distribution for the remaining pods.
+
+In order to solve this problem and to make more accurate decisions in scheduling, we added a new named
+`matchLabelKeys` to `topologySpreadConstraints`. `matchLabelKeys` is a list of pod label keys to select
+the pods over which spreading will be calculated. The keys are used to lookup values from the pod labels,
+those key-value labels are ANDed with `labelSelector` to select the group of existing pods over
+which spreading will be calculated for the incoming pod.
+
+With `matchLabelKeys`, you don't need to update the `pod.spec` between different revisions.
```

Comment:
It's not explained why you would do that. When describing the original problem (two paragraphs above) you can say that an alternative would be to add a different label value every time the deployment changes.

Reply:
OK, yeah. I add this in the paragraphs above.